helix-view/src/document.rs

```diff
@@ -139,47 +141,96 @@ pub fn canonicalize_path(path: &Path) -> std::io::Result<PathBuf> {
 use helix_lsp::lsp;
 use url::Url;
 
-impl Document {
-    pub fn new(text: Rope) -> Self {
-        let changes = ChangeSet::new(&text);
-        let old_state = None;
+// Temporary
+#[inline(always)]
+fn checked_add(num: usize, opt: Option<usize>) -> Option<usize> {
+    if let Some(n) = opt {
+        n.checked_add(num)
+    } else {
+        None
+    }
+}
 
-        Self {
-            id: DocumentId::default(),
-            path: None,
-            text,
-            selections: HashMap::default(),
-            mode: Mode::Normal,
-            restore_cursor: false,
-            syntax: None,
-            language: None,
-            changes,
-            old_state,
-            diagnostics: Vec::new(),
-            version: 0,
-            history: Cell::new(History::default()),
-            last_saved_revision: 0,
-            language_server: None,
+// Temporary
+#[inline(always)]
+fn checked_next_power_of_two(opt: Option<usize>) -> Option<usize> {
```

Comment:
Just use option.map(||..) in the code

Reply:
Yeah, I'll remove all these functions. I just have them right now since they're from `encoding_rs`.